src/graph_ir/low_graph_ir.h

```diff
@@ -0,0 +1,60 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+
+/*
+ * This is a small low-level IR that contains particles with flattened 
+ * input/output access path lists. It can be translated directly into datalog
+ * relations that similarly represent particles with flattened lists of full 
+ * access path names.
+ *
+ * There is likely a higher-level IR from which this compiles, and the higher 
+ * level IR can be generated by deserializing protobufs.
+ */
+
+
+#include <string>
+#include <vector>
+#include <set>
+
+#ifndef LOW_GRAPH_IR_
+#define LOW_GRAPH_IR_
+
+using namespace std;
+
+class LowGraphAccessPath {
```

Comment:
Why don't we call it `AccessPath`?

Reply:
It will be easier to maintain if we have a single `AccessPath` class that gives us a flattened view as needed. It is fine to starting with a flattened representation and transparently adding the nesting and hierarchy later. For example,
```
class AccessPath {
  AccessPath(const std::string& full_path_name) : full_path_name_(full_path_name) {}
  const std::string& full_path_name() const { return full_path_name_; }
  ... 
}
```

Later, we can do the following:

```
class AccessPath {
  AccessPath(const Root& root, const std::set<Selector>& selectors) ...
  const std::string& full_path_name() const { 
     // either return a pre-computed full_path or compute from the internal state.
     return full_path_name_; 
  }
  ... 
}
```

Reply:
OK, doing it the way you suggested (having one set of classes for all layers of graph IRs) does seem reasonable. I thought there was some value in having distinct types for different layers so that type checking would prevent a pass that was written for one layer (e.g. flattened) from being used on another layer (e.g. not flattened), but this doesn't seem too important and could get in the way. Let's have one set of classes as you said.